plugins/collections/removePipelines.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const schema = require('screwdriver-data-schema');
+const joi = require('joi');
+const idSchema = schema.models.collection.base.extract('id');
+const pipelineIdSchema = schema.models.pipeline.base.extract('id');
+const pipelineIdsSchema = joi
+    .alternatives()
+    .try(joi.array().items(pipelineIdSchema), pipelineIdSchema)
+    .required();
+const IDS_KEY = 'ids[]';
+
+module.exports = () => ({
+    method: 'DELETE',
+    path: '/collections/{id}/pipelines',
+    options: {
+        description: 'Delete one or more pipelines from a collection',
+        notes: 'Returns null if successful',
+        tags: ['api', 'collections'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+
+        handler: async (request, h) => {
+            const { collectionFactory, userFactory } = request.server.app;
+            const { username, scmContext } = request.auth.credentials;
+            const { id } = request.params;
+
+            // Fetch the collection and user models
+            return Promise.all([collectionFactory.get(request.params.id), userFactory.get({ username, scmContext })])
+                .then(([collection, user]) => {
+                    if (!collection) {
+                        throw boom.notFound(`Collection ${id} does not exist`);
+                    }
+                    if (!user) {
+                        throw boom.notFound(`User ${username} does not exist`);
+                    }
+                    if (collection.userId !== user.id) {
+                        throw boom.forbidden(`User ${username} does not own collection`);
+                    }
+
+                    // Only return specific pipelines
+                    const pipelineIdsQueryParam = request.query[IDS_KEY];
+
+                    if (pipelineIdsQueryParam) {
```

Comment:
if `IDS_KEY` is `id` instead of `ids[]`, then this code can be 

```diff
---const pipelineIdsQueryParam = request.query[IDS_KEY];
+++const pipelineIdsQueryParam = request.query.id;  

// pipelineIdsQueryParam would be be a single value of 10, if 'id=10' 
// and will convert be pipelineIdsQueryParam to an array of [10, 23] when `id=10&id=23`
```

Fun fact, while I was researching `req.query`. 

I found out that https://github.com/hapijs/hapi/blob/d00293248dd68e9c3e634f5a076c3d8fc19a2049/lib/request.js#L218 

`.query` is from node native query string, https://nodejs.org/api/url.html#class-urlsearchparams.

which uses `URLSearchParams` under the hood, https://nodejs.org/api/url.html#class-urlsearchparams

Reply:
Just want to mention here, that's why I used [`URLSearchParams`](https://github.com/screwdriver-cd/ui/pull/875/files#diff-684f2ffd6b15b53b3f34d55dd8bdf12113c7ccc172f32c94321f6ee600f64d91R360) in PR  [875](https://github.com/screwdriver-cd/ui/pull/875)